src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -520,6 +520,23 @@ private static bool TryConvertValue(Type type, string value, string path, out ob
                 return true;
             }
 
+            if (type == typeof(byte[]))
+            {
+                if (value is null)
```

Comment:
You could perhaps remove this null check and TryConvertValue for byte array would still be covered by the try block below it.

Also, to exercise this scenario, would be good to add another unit test: for the null check case, where values of `bytes` in your test is null.

Reply:
> You could remove this null check and TryConvertValue for byte array would still be covered by the try block below it.

I'm not sure I understand... are you suggesting I should catch the `ArgumentNullException` that would be thrown by `Convert.FromBase64String` ?

> Also, to exercise this scenario, would be good to add another unit test: cover null check case, where value of bytes in your test is null.

Good point!

Reply:
> I'm not sure I understand... are you suggesting I should catch the ArgumentNullException that would be thrown by Convert.FromBase64String ?

No. I meant just remove `if (value is null)` block and keep:

```c#
            if (type == typeof(byte[]))
            {
                try
                {
                    result = Convert.FromBase64String(value);
                }
                catch (FormatException ex)
                {
                    error = new InvalidOperationException(SR.Format(SR.Error_FailedBinding, path, type), ex);
                }
                return true;
            }
```

Reply:
your observation is correct. But this private method you are adding lines to, does not get called when value is null

Reply:
@maryamariyan I see. You're right, of course, I didn't realize the null check was done before calling this method.

Reply:
I removed the null check